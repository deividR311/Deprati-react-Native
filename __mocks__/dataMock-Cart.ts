export const DataMock_Cart = {
  carts: [
    {
      appliedOrderPromotions: [],
      appliedProductPromotions: [],
      appliedVouchers: [],
      calculated: true,
      code: '4809002',
      deliveryAddress: {
        cellphoneNumber: '34949494',
        cellphonePreffix: '09',
        city: 'COT-287',
        country: {
          isocode: 'EC',
          name: 'Ecuador'
        },
        defaultAddress: true,
        email: 'alexis.enriquezca@gmail.com',
        firstName: 'Alejandra ',
        formattedAddress: 'Calle 6, Casa 23, Cotopaxi, Alaquez',
        id: '8819555172375',
        idNumber: '1721480299',
        idType: 'CEDULA',
        line1: 'Calle 6',
        line2: 'Casa 23',
        otherInfo: 'Laura',
        phone: '',
        region: {
          countryIso: 'EC',
          isocode: 'EC-COT',
          isocodeShort: '193',
          name: 'Cotopaxi'
        },
        shippingAddress: true,
        town: 'Alaquez',
        visibleInAddressBook: true
      },
      deliveryItemsQuantity: 1,
      deliveryOrderGroups: [
        {
          entries: [
            {
              basePrice: {
                currencyIso: 'USD',
                formattedValue: '$2,32',
                priceType: 'BUY',
                value: 2.32
              },
              basePriceWithTaxes: {
                currencyIso: 'USD',
                formattedValue: '$2,60',
                priceType: 'BUY',
                value: 2.5984
              },
              basePriceWithoutTaxes: {
                currencyIso: 'USD',
                formattedValue: '$23,20',
                priceType: 'BUY',
                value: 23.2
              },
              deliveryMode: {
                code: 'standard'
              },
              deliveryTimeRange: '5 a 7 días laborables',
              entryNumber: 0,
              giftPacking: false,
              pickup: false,
              product: {
                acceptGiftPacking: true,
                availableForPickup: false,
                baseOptions: [
                  {
                    selected: {
                      code: '226010190802030008',
                      priceData: {
                        currencyIso: 'USD',
                        formattedValue: '$2,32',
                        priceType: 'BUY',
                        value: 2.32
                      },
                      stock: {
                        stockLevel: 11,
                        stockLevelStatus: 'inStock'
                      },
                      url: '/camiseta-estampada/p/226010190802030008',
                      variantOptionQualifiers: []
                    },
                    variantType: 'GenericVariantProduct'
                  }
                ],
                baseProduct: '000226010190802030',
                categories: [
                  {
                    code: 'T8023',
                    name: '10',
                    parentCategoryName: 'Talla',
                    url: '/10/c/T8023'
                  },
                  {
                    code: 'C0110601',
                    name: 'Blanco',
                    parentCategoryName: 'Color',
                    url: '/blanco/c/C0110601'
                  },
                  {
                    code: '0701',
                    name: 'MUJERES',
                    url: '/mujeres/c/0701'
                  }
                ],
                code: '226010190802030008',
                deliveryModes: [
                  {
                    code: 'standard'
                  },
                  {
                    code: 'pickup'
                  }
                ],
                ean: '15282579',
                images: [
                  {
                    format: 'zoom',
                    imageType: 'PRIMARY',
                    url: 'https://imagestg.deprati.com.ec/sys-master/images/h67/h07/8890805157918/15282579-0_product_515Wx772H'
                  },
                  {
                    format: 'product',
                    imageType: 'PRIMARY',
                    url: 'https://imagestg.deprati.com.ec/sys-master/images/h26/h0b/8890805288990/15282579-0_product_300Wx450H'
                  },
                  {
                    format: 'thumbnail',
                    imageType: 'PRIMARY',
                    url: 'https://imagestg.deprati.com.ec/sys-master/images/h38/h0e/8890805354526/15282579-0_product_96Wx144H'
                  },
                  {
                    format: 'cartIcon',
                    imageType: 'PRIMARY',
                    url: 'https://imagestg.deprati.com.ec/sys-master/images/h49/h11/8890805420062/15282579-0_product_65Wx97H'
                  }
                ],
                isGiftProduct: false,
                isPromotionSpecialPrice: false,
                ivaCondition: '12%',
                name: 'CAMISETA ESTAMPADA',
                purchasable: true,
                stock: {
                  stockLevel: 11,
                  stockLevelStatus: 'inStock'
                },
                url: '/camiseta-estampada/p/226010190802030008'
              },
              quantity: 1,
              subtotal: {
                currencyIso: 'USD',
                formattedValue: '$23,20',
                priceType: 'BUY',
                value: 23.2
              },
              totalDiscount: {
                currencyIso: 'USD',
                formattedValue: '$20,88',
                priceType: 'BUY',
                value: 20.88
              },
              totalPrice: {
                currencyIso: 'USD',
                formattedValue: '$2,32',
                priceType: 'BUY',
                value: 2.32
              },
              totalPriceWithoutTaxes: {
                currencyIso: 'USD',
                formattedValue: '$2,32',
                priceType: 'BUY',
                value: 2.32
              },
              updateable: true,
              volume: 0,
              weight: 0
            }
          ],
          totalPriceWithTax: {
            currencyIso: 'USD',
            formattedValue: '$2,60',
            priceType: 'BUY',
            value: 2.5999999999999996
          }
        }
      ],
      entries: [
        {
          basePrice: {
            currencyIso: 'USD',
            formattedValue: '$2,32',
            priceType: 'BUY',
            value: 2.32
          },
          basePriceWithTaxes: {
            currencyIso: 'USD',
            formattedValue: '$2,60',
            priceType: 'BUY',
            value: 2.5984
          },
          basePriceWithoutTaxes: {
            currencyIso: 'USD',
            formattedValue: '$23,20',
            priceType: 'BUY',
            value: 23.2
          },
          deliveryMode: {
            code: 'standard'
          },
          deliveryTimeRange: '5 a 7 días laborables',
          entryNumber: 0,
          giftPacking: false,
          pickup: false,
          product: {
            acceptGiftPacking: true,
            availableForPickup: false,
            baseOptions: [
              {
                selected: {
                  code: '226010190802030008',
                  priceData: {
                    currencyIso: 'USD',
                    formattedValue: '$2,32',
                    priceType: 'BUY',
                    value: 2.32
                  },
                  stock: {
                    stockLevel: 11,
                    stockLevelStatus: 'inStock'
                  },
                  url: '/camiseta-estampada/p/226010190802030008',
                  variantOptionQualifiers: []
                },
                variantType: 'GenericVariantProduct'
              }
            ],
            baseProduct: '000226010190802030',
            categories: [
              {
                code: 'T8023',
                name: '10',
                parentCategoryName: 'Talla',
                url: '/10/c/T8023'
              },
              {
                code: 'C0110601',
                name: 'Blanco',
                parentCategoryName: 'Color',
                url: '/blanco/c/C0110601'
              },
              {
                code: '0701',
                name: 'MUJERES',
                url: '/mujeres/c/0701'
              }
            ],
            code: '226010190802030008',
            deliveryModes: [
              {
                code: 'standard'
              },
              {
                code: 'pickup'
              }
            ],
            ean: '15282579',
            images: [
              {
                format: 'zoom',
                imageType: 'PRIMARY',
                url: 'https://imagestg.deprati.com.ec/sys-master/images/h67/h07/8890805157918/15282579-0_product_515Wx772H'
              },
              {
                format: 'product',
                imageType: 'PRIMARY',
                url: 'https://imagestg.deprati.com.ec/sys-master/images/h26/h0b/8890805288990/15282579-0_product_300Wx450H'
              },
              {
                format: 'thumbnail',
                imageType: 'PRIMARY',
                url: 'https://imagestg.deprati.com.ec/sys-master/images/h38/h0e/8890805354526/15282579-0_product_96Wx144H'
              },
              {
                format: 'cartIcon',
                imageType: 'PRIMARY',
                url: 'https://imagestg.deprati.com.ec/sys-master/images/h49/h11/8890805420062/15282579-0_product_65Wx97H'
              }
            ],
            isGiftProduct: false,
            isPromotionSpecialPrice: false,
            ivaCondition: '12%',
            name: 'CAMISETA ESTAMPADA',
            purchasable: true,
            stock: {
              stockLevel: 11,
              stockLevelStatus: 'inStock'
            },
            url: '/camiseta-estampada/p/226010190802030008'
          },
          quantity: 1,
          subtotal: {
            currencyIso: 'USD',
            formattedValue: '$23,20',
            priceType: 'BUY',
            value: 23.2
          },
          totalDiscount: {
            currencyIso: 'USD',
            formattedValue: '$20,88',
            priceType: 'BUY',
            value: 20.88
          },
          totalPrice: {
            currencyIso: 'USD',
            formattedValue: '$2,32',
            priceType: 'BUY',
            value: 2.32
          },
          totalPriceWithoutTaxes: {
            currencyIso: 'USD',
            formattedValue: '$2,32',
            priceType: 'BUY',
            value: 2.32
          },
          updateable: true,
          volume: 0,
          weight: 0
        }
      ],
      guid: '15901dbd-9966-4239-9a4d-fecc4f747589',
      net: true,
      orderDiscounts: {
        currencyIso: 'USD',
        formattedValue: '$0,00',
        priceType: 'BUY',
        value: 0
      },
      pickupItemsQuantity: 0,
      pickupOrderGroups: [],
      productDiscounts: {
        currencyIso: 'USD',
        formattedValue: '$0,00',
        priceType: 'BUY',
        value: 0
      },
      sellWithTaxes: {
        currencyIso: 'USD',
        formattedValue: '$2,32',
        priceType: 'BUY',
        value: 2.32
      },
      sellWithoutTaxWithoutInterest: {
        currencyIso: 'USD',
        formattedValue: '$2,31',
        priceType: 'BUY',
        value: 2.31
      },
      sellWithoutTaxes: {
        currencyIso: 'USD',
        formattedValue: '$0,00',
        priceType: 'BUY',
        value: 0
      },
      site: 'deprati',
      store: 'deprati',
      subTotal: {
        currencyIso: 'USD',
        formattedValue: '$2,32',
        priceType: 'BUY',
        value: 2.32
      },
      totalDiscounts: {
        currencyIso: 'USD',
        formattedValue: '$20,88',
        priceType: 'BUY',
        value: 20.88
      },
      totalItems: 1,
      totalPrice: {
        currencyIso: 'USD',
        formattedValue: '$2,32',
        priceType: 'BUY',
        value: 2.32
      },
      totalPriceWithTax: {
        currencyIso: 'USD',
        formattedValue: '$2,60',
        priceType: 'BUY',
        value: 2.6
      },
      totalTax: {
        currencyIso: 'USD',
        formattedValue: '$0,28',
        priceType: 'BUY',
        value: 0.28
      },
      totalVolume: 0,
      totalWeight: 0.25,
      user: {
        name: 'Alexis pruebas ENRIQUEZ',
        uid: 'alexis.enriquezca@gmail.com'
      },
      potentialOrderPromotions: [
        {
          consumedEntries: [],
          promotion: {
            code: '10012022_EGCE_PRUEBAAPP1_A',
            description: 'Version 2 Hybris - PRUEBA APP',
            endDate: '2024-01-01T04:59:59+0000',
            promotionType: 'Promoción basada en regla'
          }
        },
        {
          consumedEntries: [],
          description:
            'Completa $150 y  obten envío gratis. Aplica restricciones.',
          promotion: {
            code: '10012022_EGCE_PRUEBAAPP1_C',
            description: 'Version 2 Hybris',
            endDate: '2024-01-01T04:59:59+0000',
            promotionType: 'Promoción basada en regla'
          }
        }
      ],
      potentialProductPromotions: [],
      totalUnitCount: 1
    }
  ]
};
