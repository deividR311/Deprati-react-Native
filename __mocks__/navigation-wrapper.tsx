import React from 'react';
import { NavigationContainer } from '@react-navigation/native';
import { createNativeStackNavigator } from '@react-navigation/native-stack';
import { COLORS } from '../src/application/common';
import { ProvideWrapper } from './provider-wrapper';

jest.useFakeTimers();
jest.mock('react-native/Libraries/Animated/NativeAnimatedHelper');

const Stack = createNativeStackNavigator();

export const NavigationWrapper = ({ children, screen, initialParams }: any) => (
  <ProvideWrapper>
    <NavigationContainer>
      <Stack.Navigator
        initialRouteName="default"
        screenOptions={{
          headerShown: true,
          headerShadowVisible: false,
          headerTitleAlign: 'left',
          contentStyle: { backgroundColor: COLORS.WHITE }
        }}>
        {screen && (
          <Stack.Screen
            name="default"
            initialParams={initialParams ?? {}}
            component={screen}
          />
        )}
        {children && <Stack.Screen name="default" component={() => children} />}
      </Stack.Navigator>
    </NavigationContainer>
  </ProvideWrapper>
);
