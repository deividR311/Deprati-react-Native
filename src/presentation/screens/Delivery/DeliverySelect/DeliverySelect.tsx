import React, { useEffect } from 'react';
import {
  useFocusEffect,
  useIsFocused,
  useNavigation,
  useRoute
} from '@react-navigation/native';
import TemplatePage from '../../../common-components/template-page';
import {
  View,
  Text,
  StyleSheet,
  TouchableOpacity,
  Image,
  ScrollView
} from 'react-native';
import {
  COLORS,
  FontStyles,
  MARING_HORIZONTAL,
  NAV
} from '../../../../application/common';
import { DeliveryNavigationRoute } from '../../../navigation/delivery';
import CarouselAgency from '../components/CarouselAgency';
import { useLocalStorage } from '../../../../application/state-manager/services/localstorage/useLocalStorage';
import { LocalStorageKey } from '../../../../application/state-manager/services/localstorage';
import { testProps } from '../../../../application/utils/accessibleId';
import {
  useAppDispatch,
  useAppSelector
} from '../../../../application/state-manager';
import {
  setCartInfo,
  setShowLoadingScreen,
  ShoppingCartStateSelector
} from '../../../../application/state-manager/services/checkout';
import { useLazyGetShoppingCartRequest } from '../../../../infrastucture/apis/shopping-cart';
import { useAddressRemoveCartRequest } from '../../../../infrastucture/apis/delivery-address';

export default function DeliverySelect() {
  const navigation = useNavigation();
  const route = useRoute();
  const { enableContinueButton } = (route?.params as any) ?? {};
  const dispatch = useAppDispatch();

  const dataCart = useAppSelector(ShoppingCartStateSelector);
  const { deliveryAddress, code: cartId } = dataCart;

  const {
    localStorageData: { [LocalStorageKey.UserEmail]: username }
  } = useLocalStorage();

  const [handleRemoveCartAddress] = useAddressRemoveCartRequest();
  const [getCart] = useLazyGetShoppingCartRequest();

  const removeAddress = async () => {
    dispatch(setShowLoadingScreen(true));
    await handleRemoveCartAddress({
      cartId: cartId ?? '',
      username
    });
    const { data: cart, error } = await getCart({
      cartId,
      username
    });
    cart && dispatch(setCartInfo(cart));
    console.log('>>> Error loading cart', error);
    dispatch(setShowLoadingScreen(false));
  };

  useFocusEffect(
    React.useCallback(() => {
      if (!deliveryAddress) return;
      removeAddress();
    }, [deliveryAddress, username])
  );

  const focused = useIsFocused();
  useEffect(() => {
    focused && enableContinueButton(false);
  }, [focused]);

  const optionsDelivery = [
    {
      id: 1,
      icon: require('../../../../../assets/images/delivery/deliveryHome.png'),
      title: 'A DOMICILIO',
      descripcion:
        'Recibe tu pedido en tu casa, oficina o en la direcciÃ³n que nos indiques.',
      link: DeliveryNavigationRoute.DeliveryAddress
    },
    {
      id: 2,
      icon: require('../../../../../assets/images/delivery/deliveryAgency.png'),
      title: 'EN AGENCIAS AUTORIZADAS',
      descripcion:
        'Retira tu pedido en nuestras agencias autorizadas a nivel nacional.',
      link: DeliveryNavigationRoute.DeliveryThirdParty
    }
  ];

  const handleGoToLink = (screen: string) => {
    navigation.push(NAV.DELIVERY, {
      screen: screen,
      params: {
        cartId,
        dataCart
      }
    });
  };

  return (
    <TemplatePage loading={false} skeleton={null} error={false}>
      <ScrollView
        showsVerticalScrollIndicator={false}
        contentContainerStyle={{ paddingBottom: 190 }}>
        <Text style={styles.deliverySelect_title}>
          Elige como quieres recibir tu pedido
        </Text>
        <View style={styles.deliverySelect_options}>
          {optionsDelivery.map((option, index) => {
            return (
              <TouchableOpacity
                accessible
                {...testProps(`delivery_select${index}`)}
                style={styles.deliverySelect_card}
                key={`option${index}`}
                onPress={() => handleGoToLink(option.link)}>
                <View style={styles.deliverySelect_card_header}>
                  <View style={styles.deliverySelect_card_header_circle}>
                    <Image
                      source={option.icon}
                      style={styles.deliverySelect_card_header_img}
                    />
                  </View>
                </View>
                <View style={styles.deliverySelect_card_body}>
                  <Text style={styles.deliverySelect_card_body_title}>
                    {option.title}
                  </Text>
                  <Text style={styles.deliverySelect_card_body_descripcion}>
                    {option.descripcion}
                  </Text>
                  {option.id === 2 && <CarouselAgency />}
                </View>
              </TouchableOpacity>
            );
          })}
        </View>
      </ScrollView>
    </TemplatePage>
  );
}

const styles = StyleSheet.create({
  deliverySelect_title: {
    ...FontStyles.Body_1,
    textAlign: 'center',
    marginTop: 24
  },
  deliverySelect_options: {
    flexDirection: 'row',
    paddingHorizontal: MARING_HORIZONTAL,
    alignItems: 'stretch',
    justifyContent: 'space-around',
    minHeight: 320,
    marginTop: MARING_HORIZONTAL,
    flexWrap: 'wrap'
  },
  deliverySelect_card: {
    //...SHADOW_CARD,
    shadowColor: 'rgba(0, 0, 0, 0.12)',
    width: '45%',
    maxWidth: 175,
    height: '100%',
    //marginHorizontal: 8,
    borderRadius: 8,
    borderColor: 'rgba(0, 0, 0, 0.12)',
    borderWidth: 1,
    paddingBottom: 8
  },
  deliverySelect_card_header: {
    height: 93,
    width: '100%',
    alignItems: 'center',
    backgroundColor: '#E0E0E0',
    justifyContent: 'center',
    alignContent: 'center',
    borderTopLeftRadius: 7,
    borderTopRightRadius: 7
  },
  deliverySelect_card_header_circle: {
    backgroundColor: COLORS.BACKGROUNDICON,
    borderRadius: 50,
    overflow: 'hidden',
    padding: 12
  },
  deliverySelect_card_header_img: {},
  deliverySelect_card_body: {
    paddingHorizontal: 16,
    paddingVertical: 22
  },
  deliverySelect_card_body_title: {
    ...FontStyles.Subtitle,
    color: COLORS.DARKBRAND,
    textAlign: 'left'
  },
  deliverySelect_card_body_descripcion: {
    ...FontStyles.Body_2,
    textAlign: 'left',
    marginTop: 16,
    marginBottom: 8
  }
});
