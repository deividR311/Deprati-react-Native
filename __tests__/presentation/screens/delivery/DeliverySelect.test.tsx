import React from 'react';
import { render } from '@testing-library/react-native';
import { NavigationWrapper } from '../../../../__mocks__/navigation-wrapper';
import DeliverySelect from '../../../../src/presentation/screens/Delivery/DeliverySelect/DeliverySelect';

describe('DeliverySelect', () => {
  const mockInitParams = {
    dataCart: {},
    setTitleContinueButton: jest.fn(),
    enableContinueButton: jest.fn(),
    showActivityIndicatorContinueButton: jest.fn(),
    showTotalsBand: jest.fn(),
    setCurrentStep: jest.fn()
  };

  it('renders correctly in Snapshot', () => {
    const screenTest = render(
      <NavigationWrapper
        screen={DeliverySelect}
        initialParams={mockInitParams}
      />
    );
    expect(screenTest?.toJSON()).toMatchSnapshot();
  });
});
