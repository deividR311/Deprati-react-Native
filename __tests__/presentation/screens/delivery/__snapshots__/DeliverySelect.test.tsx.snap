// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DeliverySelect renders correctly in Snapshot 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    Array [
      Object {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <RNSScreenStack
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <RNSScreen
      collapsable={false}
      forwardedRef={[Function]}
      gestureResponseDistance={
        Object {
          "bottom": -1,
          "end": -1,
          "start": -1,
          "top": -1,
        }
      }
      onAppear={[Function]}
      onDisappear={[Function]}
      onDismissed={[Function]}
      onTransitionProgress={[Function]}
      onWillDisappear={[Function]}
      replaceAnimation="push"
      stackPresentation="push"
      style={
        Object {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
      swipeDirection="horizontal"
    >
      <RNSScreenStackHeaderConfig
        backButtonInCustomView={false}
        backgroundColor="rgb(255, 255, 255)"
        color="rgb(0, 122, 255)"
        direction="ltr"
        disableBackButtonMenu={false}
        hidden={false}
        hideBackButton={false}
        hideShadow={true}
        largeTitleHideShadow={false}
        title="default"
        titleColor="rgb(28, 28, 30)"
        topInsetEnabled={false}
        translucent={false}
      />
      <View
        style={
          Array [
            Object {
              "flex": 1,
            },
            Object {
              "backgroundColor": "rgb(242, 242, 242)",
            },
            Object {
              "backgroundColor": "#ffffff",
            },
          ]
        }
      >
        <RCTSafeAreaView
          emulateUnlessSupported={true}
          style={
            Object {
              "flex": 1,
            }
          }
        >
          <RCTScrollView
            contentContainerStyle={
              Object {
                "paddingBottom": 190,
              }
            }
            showsVerticalScrollIndicator={false}
          >
            <View>
              <Text
                style={
                  Object {
                    "color": "#000000",
                    "fontFamily": "Roboto-Regular",
                    "fontSize": 16,
                    "fontStyle": "normal",
                    "fontWeight": "400",
                    "letterSpacing": 0.44,
                    "lineHeight": 24,
                    "marginTop": 24,
                    "textAlign": "center",
                  }
                }
              >
                Elige como quieres recibir tu pedido
              </Text>
              <View
                style={
                  Object {
                    "alignItems": "stretch",
                    "flexDirection": "row",
                    "flexWrap": "wrap",
                    "justifyContent": "space-around",
                    "marginTop": 16,
                    "minHeight": 320,
                    "paddingHorizontal": 16,
                  }
                }
              >
                <View
                  accessibilityLabel="delivery_select0"
                  accessible={true}
                  collapsable={false}
                  focusable={true}
                  onClick={[Function]}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    Object {
                      "borderColor": "rgba(0, 0, 0, 0.12)",
                      "borderRadius": 8,
                      "borderWidth": 1,
                      "height": "100%",
                      "maxWidth": 175,
                      "opacity": 1,
                      "paddingBottom": 8,
                      "shadowColor": "rgba(0, 0, 0, 0.12)",
                      "width": "45%",
                    }
                  }
                  testID="delivery_select0"
                >
                  <View
                    style={
                      Object {
                        "alignContent": "center",
                        "alignItems": "center",
                        "backgroundColor": "#E0E0E0",
                        "borderTopLeftRadius": 7,
                        "borderTopRightRadius": 7,
                        "height": 93,
                        "justifyContent": "center",
                        "width": "100%",
                      }
                    }
                  >
                    <View
                      style={
                        Object {
                          "backgroundColor": "#C4C4C4",
                          "borderRadius": 50,
                          "overflow": "hidden",
                          "padding": 12,
                        }
                      }
                    >
                      <Image
                        source={
                          Object {
                            "testUri": "../../../assets/images/delivery/deliveryHome.png",
                          }
                        }
                        style={Object {}}
                      />
                    </View>
                  </View>
                  <View
                    style={
                      Object {
                        "paddingHorizontal": 16,
                        "paddingVertical": 22,
                      }
                    }
                  >
                    <Text
                      style={
                        Object {
                          "color": "#C22431",
                          "fontFamily": "Roboto-Medium",
                          "fontSize": 14,
                          "fontWeight": "500",
                          "letterSpacing": 0.1,
                          "textAlign": "left",
                        }
                      }
                    >
                      A DOMICILIO
                    </Text>
                    <Text
                      style={
                        Object {
                          "color": "#000000",
                          "fontFamily": "Roboto-Regular",
                          "fontSize": 14,
                          "fontWeight": "400",
                          "letterSpacing": 0.25,
                          "lineHeight": 20,
                          "marginBottom": 8,
                          "marginTop": 16,
                          "textAlign": "left",
                        }
                      }
                    >
                      Recibe tu pedido en tu casa, oficina o en la direcci√≥n que nos indiques.
                    </Text>
                  </View>
                </View>
                <View
                  accessibilityLabel="delivery_select1"
                  accessible={true}
                  collapsable={false}
                  focusable={true}
                  onClick={[Function]}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    Object {
                      "borderColor": "rgba(0, 0, 0, 0.12)",
                      "borderRadius": 8,
                      "borderWidth": 1,
                      "height": "100%",
                      "maxWidth": 175,
                      "opacity": 1,
                      "paddingBottom": 8,
                      "shadowColor": "rgba(0, 0, 0, 0.12)",
                      "width": "45%",
                    }
                  }
                  testID="delivery_select1"
                >
                  <View
                    style={
                      Object {
                        "alignContent": "center",
                        "alignItems": "center",
                        "backgroundColor": "#E0E0E0",
                        "borderTopLeftRadius": 7,
                        "borderTopRightRadius": 7,
                        "height": 93,
                        "justifyContent": "center",
                        "width": "100%",
                      }
                    }
                  >
                    <View
                      style={
                        Object {
                          "backgroundColor": "#C4C4C4",
                          "borderRadius": 50,
                          "overflow": "hidden",
                          "padding": 12,
                        }
                      }
                    >
                      <Image
                        source={
                          Object {
                            "testUri": "../../../assets/images/delivery/deliveryAgency.png",
                          }
                        }
                        style={Object {}}
                      />
                    </View>
                  </View>
                  <View
                    style={
                      Object {
                        "paddingHorizontal": 16,
                        "paddingVertical": 22,
                      }
                    }
                  >
                    <Text
                      style={
                        Object {
                          "color": "#C22431",
                          "fontFamily": "Roboto-Medium",
                          "fontSize": 14,
                          "fontWeight": "500",
                          "letterSpacing": 0.1,
                          "textAlign": "left",
                        }
                      }
                    >
                      EN AGENCIAS AUTORIZADAS
                    </Text>
                    <Text
                      style={
                        Object {
                          "color": "#000000",
                          "fontFamily": "Roboto-Regular",
                          "fontSize": 14,
                          "fontWeight": "400",
                          "letterSpacing": 0.25,
                          "lineHeight": 20,
                          "marginBottom": 8,
                          "marginTop": 16,
                          "textAlign": "left",
                        }
                      }
                    >
                      Retira tu pedido en nuestras agencias autorizadas a nivel nacional.
                    </Text>
                    <View
                      style={
                        Object {
                          "backgroundColor": "#ffffff",
                          "height": 40,
                          "width": "100%",
                        }
                      }
                    >
                      <RCTScrollView
                        activeAnimationOptions={null}
                        activeAnimationType="timing"
                        activeSlideAlignment="center"
                        activeSlideOffset={20}
                        apparitionDelay={0}
                        automaticallyAdjustContentInsets={false}
                        autoplay={true}
                        autoplayDelay={1000}
                        autoplayInterval={2000}
                        callbackOffsetMargin={5}
                        collapsable={false}
                        containerCustomStyle={Object {}}
                        contentContainerCustomStyle={Object {}}
                        contentContainerStyle={
                          Array [
                            Object {
                              "paddingLeft": 0,
                              "paddingRight": 0,
                            },
                            Object {},
                          ]
                        }
                        data={
                          Array [
                            "https://imagesdevapp.deprati.com.ec/checkout/courier1.svg",
                            "https://imagesdevapp.deprati.com.ec/checkout/courier2.svg",
                            "https://imagesdevapp.deprati.com.ec/checkout/courier1.svg",
                            "https://imagesdevapp.deprati.com.ec/checkout/courier2.svg",
                            "https://imagesdevapp.deprati.com.ec/checkout/courier1.svg",
                            "https://imagesdevapp.deprati.com.ec/checkout/courier2.svg",
                          ]
                        }
                        decelerationRate="fast"
                        directionalLockEnabled={true}
                        enableMomentum={false}
                        enableSnap={true}
                        firstItem={0}
                        forwardedRef={[Function]}
                        getItem={[Function]}
                        getItemCount={[Function]}
                        hasParallaxImages={false}
                        horizontal={true}
                        inactiveSlideOpacity={0.7}
                        inactiveSlideScale={0.9}
                        inactiveSlideShift={0}
                        initialNumToRender={6}
                        invertStickyHeaders={false}
                        inverted={false}
                        itemWidth={234.375}
                        keyExtractor={[Function]}
                        layout="default"
                        lockScrollTimeoutDuration={1000}
                        lockScrollWhileSnapping={false}
                        loop={true}
                        loopClonesPerSide={2}
                        maxToRenderPerBatch={13}
                        onContentSizeChange={[Function]}
                        onLayout={[Function]}
                        onMomentumScrollBegin={[Function]}
                        onMomentumScrollEnd={[Function]}
                        onResponderRelease={[Function]}
                        onScroll={[Function]}
                        onScrollBeginDrag={[Function]}
                        onScrollEndDrag={[Function]}
                        onStartShouldSetResponderCapture={[Function]}
                        onTouchEnd={[Function]}
                        onTouchStart={[Function]}
                        overScrollMode="never"
                        pinchGestureEnabled={false}
                        removeClippedSubviews={false}
                        renderItem={[Function]}
                        scrollEnabled={true}
                        scrollEventThrottle={1}
                        scrollsToTop={false}
                        shouldOptimizeUpdates={true}
                        showsHorizontalScrollIndicator={false}
                        showsVerticalScrollIndicator={false}
                        slideStyle={Object {}}
                        sliderWidth={234.375}
                        stickyHeaderIndices={Array []}
                        style={
                          Object {
                            "flexDirection": "row",
                            "opacity": 0,
                            "width": 234.375,
                          }
                        }
                        swipeThreshold={20}
                        useScrollView={false}
                        vertical={false}
                        viewabilityConfigCallbackPairs={Array []}
                        windowSize={13}
                      >
                        <View>
                          <View
                            onLayout={[Function]}
                            style={
                              Array [
                                Object {
                                  "flexDirection": "row",
                                },
                                null,
                              ]
                            }
                          >
                            <View
                              collapsable={false}
                              pointerEvents="box-none"
                              style={
                                Object {
                                  "opacity": 1,
                                  "transform": Array [
                                    Object {
                                      "scale": 1,
                                    },
                                  ],
                                  "width": 234.375,
                                }
                              }
                            />
                          </View>
                          <View
                            onLayout={[Function]}
                            style={
                              Array [
                                Object {
                                  "flexDirection": "row",
                                },
                                null,
                              ]
                            }
                          >
                            <View
                              collapsable={false}
                              pointerEvents="box-none"
                              style={
                                Object {
                                  "opacity": 0.7,
                                  "transform": Array [
                                    Object {
                                      "scale": 0.9,
                                    },
                                  ],
                                  "width": 234.375,
                                }
                              }
                            />
                          </View>
                          <View
                            onLayout={[Function]}
                            style={
                              Array [
                                Object {
                                  "flexDirection": "row",
                                },
                                null,
                              ]
                            }
                          >
                            <View
                              collapsable={false}
                              pointerEvents="box-none"
                              style={
                                Object {
                                  "opacity": 0.7,
                                  "transform": Array [
                                    Object {
                                      "scale": 0.9,
                                    },
                                  ],
                                  "width": 234.375,
                                }
                              }
                            />
                          </View>
                          <View
                            onLayout={[Function]}
                            style={
                              Array [
                                Object {
                                  "flexDirection": "row",
                                },
                                null,
                              ]
                            }
                          >
                            <View
                              collapsable={false}
                              pointerEvents="box-none"
                              style={
                                Object {
                                  "opacity": 0.7,
                                  "transform": Array [
                                    Object {
                                      "scale": 0.9,
                                    },
                                  ],
                                  "width": 234.375,
                                }
                              }
                            />
                          </View>
                          <View
                            onLayout={[Function]}
                            style={
                              Array [
                                Object {
                                  "flexDirection": "row",
                                },
                                null,
                              ]
                            }
                          >
                            <View
                              collapsable={false}
                              pointerEvents="box-none"
                              style={
                                Object {
                                  "opacity": 0.7,
                                  "transform": Array [
                                    Object {
                                      "scale": 0.9,
                                    },
                                  ],
                                  "width": 234.375,
                                }
                              }
                            />
                          </View>
                          <View
                            onLayout={[Function]}
                            style={
                              Array [
                                Object {
                                  "flexDirection": "row",
                                },
                                null,
                              ]
                            }
                          >
                            <View
                              collapsable={false}
                              pointerEvents="box-none"
                              style={
                                Object {
                                  "opacity": 0.7,
                                  "transform": Array [
                                    Object {
                                      "scale": 0.9,
                                    },
                                  ],
                                  "width": 234.375,
                                }
                              }
                            />
                          </View>
                        </View>
                      </RCTScrollView>
                    </View>
                  </View>
                </View>
              </View>
            </View>
          </RCTScrollView>
          <Modal
            hardwareAccelerated={false}
            transparent={true}
            visible={false}
          />
        </RCTSafeAreaView>
      </View>
    </RNSScreen>
  </RNSScreenStack>
</RNCSafeAreaProvider>
`;
